### PR TITLE
EDU-136 Test batch and task processors

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-136

### CODE REVIEW

File: src/services/batch-processor.spec.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      it('should return false', () => {
+        expect(result).toBe(false);
+      });
+    });
+
+    describe('when there are no tasks to process', () => {
+      let uncompletedBatch;
+
+      beforeEach(async () => {
+        uncompletedBatch = {};
```
Comment (ahelmberger) [inline src/services/batch-processor.spec.js]: While reading this, I think I found an error in our implementation (or maybe I have a knot in my brain?): After we have found an uncomplete batch, we try to get a random task for it **without** filtering by `batchId` (`taskStore.findRandomOne({ processed: false })`, from then on we process this task in connection with the uncomplete batch (e.g. by handling down the batch params to the concrete task processor. But what if there are two uncomplete batches and the randomly fetched task does not really belong to the batch we queried before? If I'm not mistaken, than we have to filter the random sample query by `batchId` :thinking:

File: src/services/task-processor.js [hunk 2]
```diff
@@ -71,10 +71,10 @@ export default class TaskProcessor {
       }
 
       logger.debug('Saving task');
-      this.taskStore.save(nextTask);
+      await this.taskStore.save(nextTask);
 
     } finally {
-      this.taskLockStore.releaseLock(lock);
+      await this.taskLockStore.releaseLock(lock);
```
Comment (laurentiu-ilici) [inline src/services/task-processor.js:77]: Very important await :)

File: src/services/batch-processor.spec.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+        result = await sut.process(ctx);
+      });
+
+      it('should call taskStore.findRandomOne', () => {
+        sinon.assert.calledWith(taskStore.findRandomOne, { processed: false });
+      });
+
+      it('should complete the batch', () => {
+        uncompletedBatch.completedOn = now;
```
Comment (ahelmberger) [inline src/services/batch-processor.spec.js:89]: I don't understand why we set this property here in the test setup, shouldn't we rather check if it was set by the processor?

**Review by laurentiu-ilici — APPROVED** [review]

File: src/services/batch-processor.spec.js (hunk 1 above)
Comment (laurentiu-ilici) [reply src/services/batch-processor.spec.js]: I lingered on the same thought a few minutes ago :))

Comment (adrianaluchian) [reply src/services/batch-processor.spec.js]: Very good catch, and I think we might have mentioned this while implementing and forgot to actually do it 😬 Will correct!

File: src/services/task-processor.js (hunk 2 above)
Comment (adrianaluchian) [reply src/services/task-processor.js:77]: Oh there were even 2 of them that we had missed 😅

File: src/services/batch-processor.spec.js (hunk 3 above)
Comment (adrianaluchian) [reply src/services/batch-processor.spec.js]: Oh because I got carried away and this is wrong, corrected!

**Review by ahelmberger — APPROVED** [review]

**Pull request merged**